ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -309,24 +309,28 @@ sealed abstract class Queries(tablePrefix: String, tpIdCacheMaxEntries: Long)(im
       cids: Map[SurrogateTpId, Set[String]]
   )(implicit log: LogHandler): ConnectionIO[Int] = {
     import cats.instances.vector._
-    import cats.instances.int._
-    import cats.syntax.foldable._
     import nonempty.catsinstances._
     (cids: Iterable[(SurrogateTpId, Set[String])]).view.collect { case (k, NonEmpty(cids)) =>
       (k, cids)
     }.toMap match {
       case NonEmpty(cids) =>
-        val chunks = maxListSize.fold(Vector(cids))(chunkBySetSize(_, cids))
-        chunks.map { chunk =>
-          (fr"DELETE FROM $contractTableName WHERE " ++
-            joinFragment(
-              chunk.toVector.map { case (tpid, cids) =>
-                val inCids = Fragments.in(fr"contract_id", cids.toVector.toNEF)
-                fr"(tpid = $tpid AND $inCids)"
-              }.toOneAnd,
-              fr" OR ",
-            )).update.run
-        }.foldA
+        val del = fr"DELETE FROM $contractTableName WHERE " ++ {
+          val chunkMap =
+            maxListSize.fold(cids.transform((_, sv) => NonEmpty(Vector, sv)))(
+              chunkBySetSize(_, cids)
+            )
+          joinFragment(
+            chunkMap.toVector.map { case (tpid, chunks) =>
```

Comment:
Very minor, but is `toVector` needed?

Reply:
I got rid of a few minor needless conversions in this PR, no reason we can't garden some more later as we rework `joinFragment` and its utilities a bit.